python_modules/dagster/dagster/_core/secrets/env_file.py

```diff
@@ -0,0 +1,58 @@
+import logging
+import os
+from typing import Dict, Optional
+
+from dotenv import dotenv_values
+
+import dagster._check as check
+from dagster._config import Field, StringSource
+from dagster._serdes import ConfigurableClass
+
+from .loader import SecretsLoader
+
+
+class EnvFileLoader(SecretsLoader, ConfigurableClass):
+    def __init__(
+        self,
+        inst_data=None,
+        base_dir=None,
+    ):
+        self._inst_data = inst_data
+        self._base_dir = base_dir or os.getcwd()
+
+    def _get_env_vars(self, file_path) -> Dict[str, str]:
+        env_var_dict: Dict[str, str] = {
+            key: check.not_none(val) for key, val in dotenv_values(file_path).items() if val != None
+        }
+
+        if len(env_var_dict):
+            logging.getLogger("dagster").info(
+                "Loaded environment variables from .env file: "
+                + ",".join([env_var for env_var in env_var_dict]),
+            )
+        else:
+            logging.getLogger("dagster").info("No environment variables in .env file")
+
+        return env_var_dict
+
+    def get_secrets_for_environment(self, location_name: Optional[str]) -> Dict[str, str]:
+        env_vars: Dict[str, str] = {}
+
+        env_file_path = os.path.join(self._base_dir, ".env")
+
+        if os.path.exists(env_file_path):
```

Comment:
@petehunt we can and we do!

Reply:
oh i see! so it's not that easy to discover this feature from loading the code right? 
i think it's totally fine and in fact great as we're still early, but eventually, i'd like to hint the existence of this behavior by default (once we have the docs ready to link)